### PR TITLE
U-Net Code, did not have a original research paper, attached along wi…

### PR DESCRIPTION
**Pull request by NiharJani2002**
…th code, added original research paper, followed the way keras-cv adds research paper along with code

# What does this PR do?
This PR Adds original research paper of U-Net In The Code (unet.py)

Fixes # (issue)
In Continuation with Pull Request #1727 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
Ans) YES
Improves The DOCSTRING Of U-Net (unet.py)
Followed the Standard Convention of Adding Research Paper as seen from previous examples  Keras-CV repo.

## Who can review?
@ianstenbit

### CODE REVIEW

File: keras_cv/models/__internal__/unet.py [hunk 1]
```diff
@@ -177,6 +177,12 @@ def UNet(
     All function parameters require curried functions as inputs which return a
     function that acts on tensors as inputs.
 
+    Reference:
+        - [U-Net: Convolutional Networks for Biomedical Image Segmentation](https://arxiv.org/abs/1505.04597)
+
+            U-Net relies on strong use of Data Augmentation and architecture consists of a contracting path to capture
```
Comment (ianstenbit) [inline keras_cv/models/__internal__/unet.py:183]: I think we can drop this second sentence, but the reference part LGTM -- thanks!

**Review by ianstenbit — APPROVED** [review]
Thanks for the PR! Looks good after one small change

File: keras_cv/models/__internal__/unet.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/models/__internal__/unet.py:183]: Sorry I think I wasn't clear -- let's only include the link (and drop the part that starts with "U-Net relies on strong use ..."

Comment (NiharJani2002) [inline keras_cv/models/__internal__/unet.py:183]:
```suggestion
       
```

**Review by NiharJani2002 — COMMENTED** [review]
Per Your Latest Suggestion, Now Research Paper Is There In The Code.
"U-Net relies on strong...." is removed.

File: keras_cv/models/__internal__/unet.py [hunk 2]
```diff
@@ -177,6 +177,12 @@ def UNet(
     All function parameters require curried functions as inputs which return a
     function that acts on tensors as inputs.
 
+    Reference:
+        - [U-Net: Convolutional Networks for Biomedical Image Segmentation](https://arxiv.org/abs/1505.04597)
+
```
Comment (ianstenbit) [inline keras_cv/models/__internal__/unet.py:182]: We ended up with a lot of newlines here, let's just have one newline between the reference link and the `Args` section. Then I can merge 👍

Comment (NiharJani2002) [reply keras_cv/models/__internal__/unet.py]: Removed The Extra Lines Between The Research Paper And The Code

**Comment by ianstenbit** [comment]
@NiharJani2002 The linter is unhappy. You can run `sh shell/format.sh` to fix most issues locally and then run `sh shell/lint.sh` to verify that the code format is correct. In this case, we'll need to add `  # noqa: E501` to the end of the model docstring to make the linter happy.

**Comment by NiharJani2002** [comment]
I will check This out (Linter), and generate a new Pull Request, Till Then I am closing this pull request.,
Thanks for the new tool Linter.

**Pull request closed without merging**